qlib/kernel/asm/aarch64.rs

```diff
@@ -524,3 +535,30 @@ pub unsafe fn midr() -> u64 {
     asm!("mrs {0}, midr_el1", out(reg) ret);
     ret
 }
+
+
+pub unsafe fn sp_el0() -> u64 {
```

Comment:
1. This function should be annotated with `[inline]`, to be on the safe side that the compiler will not create a new frame.
2. Is this nested call actually needed? `asm!("...", ...);` could be placed directly inside `pub fn GetCurrentUserSp() -> u64`

Reply:
Actually there is no place to call the GetCurrentUserSp until now, And I think we can make all functions of the read of a register with it's name and write of the register with the form of `<register_name>_write`, so that it is in such a unified way. The GetCurrentUserSp is a function name to let the callers knows what they are doing. 

And I think as the function is annotated inline, compiler will do the optimization so we don't have to care about the overhead of extra function frame.

Reply:
> This function should be annotated with [inline], to be on the safe side that the compiler will not create a new frame.

added